pkg/parser/external/java_parser.go

```diff
@@ -34,56 +32,69 @@ func newJavaParser(typ string) *javaParser {
 	return &javaParser{typ: typ}
 }
 
-type parseResult struct {
-	Statements []string `json:"statements"`
-	Position   int      `json:"position"`
-	Error      string   `json:"error"`
+func getServerAddress() string {
+	ip := getEnv("SQLFLOW_EXTERNAL_PARSER_IP", "localhost")
```

Comment:
I suggest we keep only the port environment variable and remove the IP one.

Referring to https://golang.org/pkg/net/?m=all#Dial, we see that this function could return `fmt.Sprintf(":%d", port)` without the IP.  In this case, the Go net package tries to connect to the port bound to any of the network interfaces.

Reply:
Sounds good.